src/Scripting/Core/ScriptSourceResolver.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Scripting
+{
+    using static ParameterValidationHelpers;
+
+    public sealed class ScriptSourceResolver : SourceFileResolver, IEquatable<ScriptSourceResolver>
```

Comment:
I haven't bothered to check the references, but does this need to be public?  It feels like an implementation detail that people could trivially duplicate if they're wanting to implement their own host.